src/components/Button/MucButton.vue

```diff
@@ -1,18 +1,56 @@
-<script setup lang="ts">
-const props = defineProps({
-  expanded: {
-    type: Boolean,
-    default: false
-  }
-})
-</script>
-
 <template>
-  <button class="m-button m-button--primary m-button--animated-right">
+  <button
+    @click="handleClick"
+    :disabled="disabled"
+    class="m-button m-button--animated-right"
+    :class="buttonClass"
+  >
     <slot />
   </button>
 </template>
 
-<style scoped>
+<script setup lang="ts">
+import { computed } from "vue"
 
-</style>
\ No newline at end of file
+type buttonType = "primary" | "secondary" | "ghost"
+
+const props = withDefaults(
+  defineProps<{
+    variant?: buttonType
+    disabled?: boolean
+  }>(),
+  {
+    variant: "primary",
+    disabled: false,
+  }
+)
+
+const emit = defineEmits<{
+  (e: "click"): void
+}>()
+
+const buttonClass = computed(() => {
+  switch (props.variant) {
+    case "secondary":
+      return "m-button--secondary"
+    case "ghost":
+      return "m-button--ghost"
+    default:
+      return "m-button--primary"
+  }
+})
+
+const handleClick = () => {
+  emit("click")
+}
+</script>
+
+<style scoped>
+/*
+* Class below may look unused but is used dynamically! Do NOT remove them!!
+* It adds padding to ghost button, which is missing in the patternlab!
+*/
+.m-button--ghost {
+  padding-left: 14px;
+}
+</style>
```

Comment:
```suggestion
```

I think you use this padding because the ghost button is aligned more to the left as it has no bounding box around it. Even though I understand why you do it I would prefer not to include this padding in the component and keep it the way it's designed in the pattern lab

Reply:
css-class is removed